sys-block/gpart/patches/gpart-0.3.patchset.2

```diff
@@ -0,0 +1,132 @@
+From 91a5196b57f2e80b172313b69dd7fe0475db85d5 Mon Sep 17 00:00:00 2001
+From: My Name <mymail@mydomain.org>
+Date: Sat, 6 Jan 2018 14:00:23 +0000
+Subject: applying patch gpart-0.3.patchset.2
+
+
+diff --git a/src/disku.c b/src/disku.c
+index d50cebf..794abe9 100644
+--- a/src/disku.c
++++ b/src/disku.c
+@@ -33,6 +33,11 @@
+ #include <sys/disk.h>
+ #endif
+ 
++#if defined(__HAIKU__)
++#include <drivers/Drivers.h>
++#include <fcntl.h>
++#endif
++
+ #include <unistd.h>
+ 
+ static void geometry_from_num_sectors(struct disk_geom *g, uint64_t nsects)
+@@ -95,8 +100,22 @@ static void os_disk_geometry(disk_desc *d, struct disk_geom *g)
+ 	g.d_nsecs = o / u;
+ 	g.d_c = g.d_nsecs / g.d_h / g.d_s;
+ }
++#elif defined(__HAIKU__)
++static void os_disk_geometry(disk_desc *d, struct disk_geom *g)
++{
++	device_geometry geom;
++
++	if (ioctl(d->d_fd, B_GET_GEOMETRY, &geom, sizeof(geom)) == -1) {
++		pr(FATAL, EM_IOCTLFAILED, "B_GET_GEOMETRY", strerror(errno));
++	}
```

Comment:
[`pr()`][0] won't abort the execution flow. You should put the below code to an `else` branch instead

[0]: https://github.com/baruch/gpart/blob/deffb405040d6f70682657bcf930f92e2a9a7458/src/gpart.c#L106

Reply:
@alaviss For FATAL level it will: https://github.com/baruch/gpart/blob/deffb405040d6f70682657bcf930f92e2a9a7458/src/gpart.c#L124

The whole project is based on this logic.